src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -471,7 +471,20 @@ int makeDagTempSlot(TypeWithAnnotations type, BoundDagTemp temp)
             }
         }
 
-        public override BoundNode VisitSwitchExpression(BoundSwitchExpression node)
+        public override BoundNode VisitConvertedSwitchExpression(BoundConvertedSwitchExpression node)
+        {
+            bool inferType = node.NaturalTypeOpt is object && node.NaturalTypeOpt.Equals(node.Type, TypeCompareKind.ConsiderEverything);
+            VisitSwitchExpressionCore(node, inferType);
+            return null;
+        }
+
+        public override BoundNode VisitUnconvertedSwitchExpression(BoundUnconvertedSwitchExpression node)
```

Comment:
>VisitUnconvertedSwitchExpression [](start = 34, length = 32)

can we get an unconverted switch expression all the way to NullableWalker?

Reply:
Yes, in error scenarios.

---
In reply to: [301816928](https://github.com/dotnet/roslyn/pull/37052#discussion_r301816928) [](ancestors = 301816928)